eslint.config.js

```diff
@@ -0,0 +1,14 @@
+import eslint from '@eslint/js';
+import tseslint from 'typescript-eslint';
+
+export default tseslint.config(
+    eslint.configs.recommended,
+    ...tseslint.configs.recommended,
+    {
+        rules: {
+            // TODO: Fix these rule and remove their overrides.
```

Comment:
These will have to be fixed later, as they would mean changing public APIs, so I'll handle that in a separate PR.